experiment/compatibility-versions/validate-compatibility-versions-feature-gates.sh

```diff
@@ -255,8 +255,10 @@ for feature_name in "${!actual_features[@]}"; do
         continue
       fi
       # Check if gate is deprecated feature gate at the same version as the binary version
-      # OR if it's a GA feature with version 1.0
+      # OR if it's a BETA feature with version 1.0 for bug fix that is bound to binary version instead of emulation version
+      # OR if it's a GA feature with version 1.0 for code deprecation
       if [[ -n "${current_version_stage[$feature_name]:-}" && "${current_version_stage[$feature_name]}" == "DEPRECATED" ]] || \
```

Comment:
nit: can't these just be
```suggestion
      if [[ -n "${current_version_stage[$feature_name]:-}" == "DEPRECATED" ]] || \
```